### PR TITLE
fix: the zoom button disabled in some cases #619

### PR DESCRIPTION
**Pull request by SuokN**
# Description
Please check this decision
Fixes # the zoom button disabled in some cases #619

## Type of change

[comment]: # 'Please delete options that are not relevant.'

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots

[comment]: # 'Please include screenshots of your changes if relevant.'

|       Before        |       After        |
| :-----------------: | :----------------: |
|enabled|disabled2|

# How Has This Been Tested?

- [ ] Cypress integration
- [ ] Cypress component tests
- [ ] Jest unit tests

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/ZoomInOutButton.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { useEffect, useState } from 'react';
+import ZoomIn from '../images/zoom-in.svg';
+import ZoomOut from '../images/zoom-out.svg';
+import { useMapContext } from '../mapContext';
+
+export default function ZoomInOutButton() {
+  const mapContext = useMapContext();
+  const [isEnabledZoomIn, setIsEnabledZoomIn] = useState(false);
+  const [isEnabledZoomOut, setIsEnabledZoomOut] = useState(false);
+  const MOVE_END_EVENT = 'move-end';
```
Comment (RubenSmn) [inline src/components/ZoomInOutButton.js:12]: We can get this from the `Map` itself

```js
import { Map } from 'treetracker-web-map-core';

Map.REGISTERED_EVENTS.MOVE_END
```

File: src/components/ZoomInOutButton.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  }
+
+  useEffect(() => {
+    if (mapContext.map) {
+      mapContext.map.on(MOVE_END_EVENT, () => {
+        checkZoomEnable();
+      });
+    }
+  });
```
Comment (RubenSmn) [inline src/components/ZoomInOutButton.js:28]: We should add a dependency array here

**Review by RubenSmn — CHANGES_REQUESTED** [review]
When you're on the `/top` page the initial zoom is already the max zoom but the button does not show that its disabled yet.

**Comment by dadiorchen** [comment]
@SuokN can you help fix the CI for unit tests? I checked the detail, it's a eslint error, should be easy to fix

**Comment by dadiorchen** [comment]
@SuokN thank you!

**Pull request merged**